spec/avram/bulk_upsert_spec.cr

```diff
@@ -0,0 +1,19 @@
+require "../spec_helper"
+
+describe Avram::BulkUpsert do
+  describe "bulk upserts" do
+    # Ideally, compiler can catch this / this should be impossible..
+    context "when collections mismatch" do
+    end
+
+    context "when mixed new records and updated records" do
+      # Insert spec example.
+      it "inserts with a hash of String" do
```

Comment:
raw hashes are actually disallowed on purpose because they're not safe. They are easier to work with, but also impose security risk. This is why you can't pass a hash in to `create` or `update` https://github.com/luckyframework/avram/blob/ac281f490a85283eaf8b74fd23357dd70a6d7f22/src/avram/needy_initializer_and_save_methods.cr#L104

Reply:
Yeah this is placeholder code stolen from insert to remind me about some of the tests I want to add, need to update the tests f’sho.